### PR TITLE
Config option exclude zmq

### PR DESCRIPTION
**Pull request by reiinakano**
Thank goodness Satnogs did something similar

### CODE REVIEW

File: grc/CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +18,21 @@
 # the Free Software Foundation, Inc., 51 Franklin Street,
 # Boston, MA 02110-1301, USA.
 
-install(FILES
+list(APPEND debug_blocks
```
Comment (anuraaga) [inline grc/CMakeLists.txt]: Why use the word debug instead of zmq?

File: CMakeLists.txt [hunk 2]
```diff
@@ -149,7 +150,14 @@ if(NOT CPPUNIT_FOUND)
     message(FATAL_ERROR "CppUnit required to compile sattools")
 endif()
 
-find_package(ZeroMQ)
+########################################################################
+# Option to include zmq block
+########################################################################
+option(INCLUDE_ZMQ_BLOCK "Include blocks that need ZMQ" ON)
```
Comment (anuraaga) [inline CMakeLists.txt]: Let's use gnuradio conventions, `ENABLE_SATTOOLS_ZEROMQ`, similar to GR_ZEROMQ

https://github.com/gnuradio/gnuradio/blob/master/gr-zeromq/CMakeLists.txt

File: swig/sattools_swig0.i [hunk 3]
```diff
@@ -5,19 +5,16 @@
...
 %{
-#include "sattools/csp_zmqhub_pub_sink.h"
 #include "sattools/reed_solomon_ccsds_decode.h"
 #include "sattools/async_crc32c.h"
 #include "sattools/golay24_decode.h"
 #include "sattools/asm_golay_decoder.h"
 %}
 
 
-%include "sattools/csp_zmqhub_pub_sink.h"
```
Comment (anuraaga) [inline swig/sattools_swig0.i]: Does it not work to flag out the zeromq stuff using `#IFDEF` with a define passed into swig? Then we wouldn't need the complexity of all these files I think.

**Comment by kenichi-fukushima** [comment]
Hmm, it is quite complex. Is it so important to remove those files from Starcoder distribution?

**Comment by anuraaga** [comment]
`zeromq` is a pretty specialized thing and we'd like to be able to build starcoder without it. I think a lot of the complexity would go away if rei tries using an `#ifdef` in the swig file though, what do you think @reiinakano ?

**Comment by reiinakano** [comment]
Done. PTAL

File: swig/CMakeLists.txt [hunk 4]
```diff
@@ -44,11 +44,22 @@ foreach(incdir ${GNURADIO_RUNTIME_INCLUDE_DIRS})
     list(APPEND GR_SWIG_INCLUDE_DIRS ${incdir}/gnuradio/swig)
 endforeach(incdir)
 
+if(${ENABLE_SATTOOLS_ZEROMQ})
+    set(ENABLE_SATTOOLS_ZEROMQ_DEFINE_STATEMENT "#define _INCLUDE_ZMQ 1")
+else()
+    set(ENABLE_SATTOOLS_ZEROMQ_DEFINE_STATEMENT "")
+endif()
+
+configure_file(
```
Comment (anuraaga) [inline swig/CMakeLists.txt]: Can you use `GR_SWIG_FLAGS` instead of a template?

**Comment by kenichi-fukushima** [comment]
This version of code might be slightly simpler than the previous version but I still don't think it has value that's worth the complexity. Even if we don't use those file, overhead in compile time and distribution size should be very small. We build the whole GnuRadio in the first place, which has many blocks that we don't use. If we want to really exclude them, we should just delete them.

**Comment by anuraaga** [comment]
We do disable many blocks, including GnuRadio's ZeroMQ blocks. GnuRadio usually has flags to enable / disable major groups of blocks, and supporting it is quite idiomatic - we should be able to upstream this change.

File: swig/CMakeLists.txt (hunk 4 above)
Comment (reiinakano) [reply swig/CMakeLists.txt]: It's unclear to me how to use SWIG flags to do what we want. I will need a much deeper understanding of SWIG for this. Maybe we could have it as a TODO.

Comment (anuraaga) [reply swig/CMakeLists.txt]: I was thinking `GR_SWIG_FLAGS=-D_INCLUDE_ZMQ` would work - swig is similar to gcc and can take preprocessor defines with the same syntax from what I understand.

If not then this approach is fine.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**